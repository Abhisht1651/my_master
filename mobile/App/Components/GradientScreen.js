// @flow

import React from 'react'
import { View } from 'react-native'

// Styles
import PropTypes from 'prop-types'
import styles from './Styles/GradientStyles'
import CircularGradient from './CircularGradient'

type Props = {
  middle: PropTypes.Object
}
const GradientScreen = (props: Props) => {
  const color1 = ['rgba(70, 195, 244, 1)', 'rgba(178, 124, 243, 1)']
  const color2 = ['rgba(70, 195, 244, .8)', 'rgba(178, 124, 243, .8)']
  const color3 = ['rgba(70, 195, 244, .6)', 'rgba(178, 124, 243, .6)']
  const color4 = ['rgba(70, 195, 244, .4)', 'rgba(178, 124, 243, .4)']
  const color5 = ['rgba(70, 195, 244, .2)', 'rgba(178, 124, 243, .2)']
  const { middle } = props
  return (
    <View style={styles.section}>
      <CircularGradient
        colors={color5}
        position={styles.position}
        linearGradient={styles.circle1_Outside}
        inside={styles.circle1_Inside}
      />
      <CircularGradient
        colors={color4}
        position={styles.position}
        linearGradient={styles.circle2_Outside}
        inside={styles.circle2_Inside}
      />
      <CircularGradient
        colors={color3}
        position={styles.position}
        linearGradient={styles.circle3_Outside}
        inside={styles.circle3_Inside}
      />
      <CircularGradient
        colors={color2}
        position={styles.position}
        linearGradient={styles.circle4_Outside}
        inside={styles.circle4_Inside}
      />
      <CircularGradient
        colors={color1}
        position={styles.position}
        linearGradient={styles.circle5_Outside}
        inside={styles.circle5_Inside}
        source={middle}
      />
    </View>
  )
}
GradientScreen.defaultProps = {
  middle: undefined
}
export default GradientScreen
