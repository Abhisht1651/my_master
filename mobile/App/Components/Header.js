// @flow
import React from 'react'
import { Text, Image, View, TouchableOpacity, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
import ProgressBar from './ProgressBar/Progress'

import { Images } from '../Themes'
import styles from './Styles/HeaderStyles'

type Props = {
  container?: StyleSheet,
  onPressBack?: PropTypes.func,
  progressContainer?: StyleSheet,
  count: number,
  onPressRight?: PropTypes.func,
  rightTextStyle?: StyleSheet,
  rightText?: string,
  isProgressBarVisible: boolean
}
const Header = (props: Props) => {
  const {
    container,
    onPressBack,
    progressContainer,
    count,
    onPressRight,
    rightTextStyle,
    rightText,
    isProgressBarVisible
  } = props
  return (
    <View style={[styles.container, container]}>
      <View>
        {onPressBack && (
          <TouchableOpacity onPress={onPressBack}>
            <Image source={Images.leftArrow} style={styles.leftArrow} />
          </TouchableOpacity>
        )}
      </View>
      {isProgressBarVisible && (
        <View style={[styles.progressContainer, progressContainer]}>
          <ProgressBar count={count} />
        </View>
      )}
      <View>
        <TouchableOpacity onPress={onPressRight}>
          <Text style={[styles.rightText, rightTextStyle]}>{rightText}</Text>
        </TouchableOpacity>
      </View>
    </View>
  )
}

Header.defaultProps = {
  container: StyleSheet.create({}),
  onPressBack: undefined,
  progressContainer: StyleSheet.create({}),
  onPressRight: () => {},
  rightTextStyle: StyleSheet.create({}),
  rightText: '    ',
  count: 1,
  isProgressBarVisible: true
}

export default Header
