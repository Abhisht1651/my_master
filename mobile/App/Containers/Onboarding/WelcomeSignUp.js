// @flow
import { NavigationContainerProps } from '@react-navigation/native'
import React from 'react'
import { Text, View, Image, Platform } from 'react-native'
import { Images } from '../../Themes'

// Styles
import styles from '../Styles/Onboarding/WelcomeSignUpStyles'
import GradientScreen from '../../Components/GradientScreen'
import Header from '../../Components/Header'

type Props = NavigationContainerProps & {}
const WelcomeSignUp = (props: Props) => {
  const [hideSplash, setHideSplash] = React.useState(false)
  React.useEffect(() => {
    setTimeout(() => {
      setHideSplash(true)
    }, 5000) // amount of time the splash is shown from the time component is rendered
  }, [])
  React.useEffect(() => {
    if (hideSplash) props.navigation.navigate('NewsFeed')
  }, [hideSplash])
  return (
    <View style={styles.section}>
      {Platform.OS === 'ios' && <View style={styles.statusBar} />}
      <View style={styles.section}>
        <GradientScreen middle={<Image source={Images.newzeraPurpleLogo} style={styles.logo} />} />
        <View style={styles.header}>
          <Header count={3} />
          <Text style={styles.text}>Welcome to Newzera!</Text>
        </View>
      </View>
    </View>
  )
}

export default WelcomeSignUp
