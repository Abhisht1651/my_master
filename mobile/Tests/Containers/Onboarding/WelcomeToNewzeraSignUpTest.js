import { Text } from 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import WelcomeSignUp from '../../../App/Containers/Onboarding/WelcomeSignUp'
import GradientScreen from '../../../App/Components/GradientScreen'
import Header from '../../../App/Components/Header'

const createTestProps = () => ({
  navigation: {
    navigate: jest.fn()
  }
})

test('WelcomeSignUp component renders correctly', () => {
  const props = createTestProps({})
  const tree = renderer.create(<WelcomeSignUp props />).toJSON()
  let x = props
  if (x) x = props
  expect(tree).toMatchSnapshot()
})

describe('WelcomeSignUp ', () => {
  describe('rendering', () => {
    let wrapper
    let props // use type "any" to opt-out of type-checking
    beforeEach(() => {
      props = createTestProps({})
      wrapper = shallow(<WelcomeSignUp props />) // no compile-time error
      let x = props
      if (x) x = props
    })
    it('There should be 1 gradient screen', () => {
      expect(wrapper.find(GradientScreen)).toHaveLength(1)
    })
    it('There should be 1 header', () => {
      expect(wrapper.find(Header)).toHaveLength(1)
    })
    it('There should be 1 Text Component', () => {
      expect(wrapper.find(Text)).toHaveLength(1)
    })
    it('match Text prop text', () => {
      const grad = wrapper.find(Text).at(0)
      expect(grad.prop('children')).toBe('Welcome to Newzera!')
    })
  })
})
