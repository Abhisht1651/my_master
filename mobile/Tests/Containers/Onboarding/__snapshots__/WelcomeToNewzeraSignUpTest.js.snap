// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeSignUp component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 58.188802083333336,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <BVLinearGradient
          colors={
            Array [
              860275700,
              867335411,
            ]
          }
          endPoint={
            Object {
              "x": 0.5,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0.5,
              "y": 0,
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 250,
                "height": 500,
                "justifyContent": "center",
                "width": 500,
              },
              Object {
                "borderRadius": 663.3035714285714,
                "height": 1326.607142857143,
                "width": 1326.607142857143,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 245,
                  "height": 490,
                  "justifyContent": "center",
                  "width": 490,
                },
                Object {
                  "borderRadius": 655.2995714285714,
                  "height": 1310.5991428571429,
                  "width": 1310.5991428571429,
                },
              ]
            }
          />
        </BVLinearGradient>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <BVLinearGradient
          colors={
            Array [
              1715913716,
              1722973427,
            ]
          }
          endPoint={
            Object {
              "x": 0.5,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0.5,
              "y": 0,
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 250,
                "height": 500,
                "justifyContent": "center",
                "width": 500,
              },
              Object {
                "borderRadius": 575,
                "height": 1150,
                "width": 1150,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 245,
                  "height": 490,
                  "justifyContent": "center",
                  "width": 490,
                },
                Object {
                  "borderRadius": 566.9959999999999,
                  "height": 1133.9919999999997,
                  "width": 1133.9919999999997,
                },
              ]
            }
          />
        </BVLinearGradient>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <BVLinearGradient
          colors={
            Array [
              2571551732,
              2578611443,
            ]
          }
          endPoint={
            Object {
              "x": 0.5,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0.5,
              "y": 0,
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 250,
                "height": 500,
                "justifyContent": "center",
                "width": 500,
              },
              Object {
                "borderRadius": 454.25,
                "height": 908.5,
                "width": 908.5,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 245,
                  "height": 490,
                  "justifyContent": "center",
                  "width": 490,
                },
                Object {
                  "borderRadius": 446.246,
                  "height": 892.492,
                  "width": 892.492,
                },
              ]
            }
          />
        </BVLinearGradient>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <BVLinearGradient
          colors={
            Array [
              3427189748,
              3434249459,
            ]
          }
          endPoint={
            Object {
              "x": 0.5,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0.5,
              "y": 0,
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 250,
                "height": 500,
                "justifyContent": "center",
                "width": 500,
              },
              Object {
                "borderRadius": 338.42857142857144,
                "height": 676.8571428571429,
                "width": 676.8571428571429,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 245,
                  "height": 490,
                  "justifyContent": "center",
                  "width": 490,
                },
                Object {
                  "borderRadius": 330.4245714285714,
                  "height": 660.8491428571429,
                  "width": 660.8491428571429,
                },
              ]
            }
          />
        </BVLinearGradient>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
          }
        }
      >
        <BVLinearGradient
          colors={
            Array [
              4282827764,
              4289887475,
            ]
          }
          endPoint={
            Object {
              "x": 0.5,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0.5,
              "y": 0,
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 250,
                "height": 500,
                "justifyContent": "center",
                "width": 500,
              },
              Object {
                "borderRadius": 243.14285714285714,
                "height": 486.2857142857143,
                "width": 486.2857142857143,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 245,
                  "height": 490,
                  "justifyContent": "center",
                  "width": 490,
                },
                Object {
                  "borderRadius": 235.13885714285715,
                  "height": 470.2777142857143,
                  "width": 470.2777142857143,
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "default": "",
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "height": 226.78000000000003,
                  "marginBottom": "12.3%",
                  "resizeMode": "contain",
                  "width": 434.99999999999994,
                }
              }
            />
          </View>
        </BVLinearGradient>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "top": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 101.384,
              "justifyContent": "space-between",
              "padding": "6%",
            },
            Object {},
          ]
        }
      >
        <View />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 5,
                "justifyContent": "space-between",
                "marginLeft": 30,
                "width": 42,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": 40,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#7B46E4",
                    "borderRadius": 2,
                    "height": 2.5,
                    "width": 12,
                  }
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "#7B46E4",
                    "borderRadius": 2,
                    "height": 2.5,
                    "width": 12,
                  }
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "#7B46E4",
                    "borderRadius": 2,
                    "height": 2.5,
                    "width": 12,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#7B46E4",
                    "fontFamily": "BananaGrotesk-Bold",
                    "fontSize": 31,
                  },
                  Object {},
                ]
              }
            >
                  
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Medium",
            "fontSize": 36,
          }
        }
      >
        Welcome to Newzera!
      </Text>
    </View>
  </View>
</View>
`;
